src/hotspot/share/opto/library_call.cpp

```diff
@@ -5364,8 +5364,9 @@ bool LibraryCallKit::inline_native_hashcode(bool is_virtual, bool is_static) {
 
   if (!UseObjectMonitorTable) {
     // Test the header to see if it is safe to read w.r.t. locking.
-  // This also serves as guard against inline types
-    Node *lock_mask      = _gvn.MakeConX(markWord::inline_type_mask_in_place);
+    // We cannot use the inline type mask as this may check bits that are overriden
+    // by an object monitor's pointer when inflating locking.
+    Node *lock_mask      = _gvn.MakeConX(markWord::lock_mask_in_place);
```

Comment:
FTR, this merge commit broke it: https://github.com/openjdk/valhalla/commit/0e3418df579fa7d2c6f56b4064656a8223f362db